great_expectations/experimental/datasources/file_path_data_asset.py

```diff
@@ -134,19 +156,100 @@ def _validate_batch_request(self, batch_request: BatchRequest) -> None:
+        batch_spec_options: dict
+        batch_data: Any
+        batch_markers: BatchMarkers
+        batch_metadata: BatchRequestOptions
+        batch: Batch
+        for batch_definition in batch_definition_list:
+            batch_spec = data_connector.build_batch_spec(
+                batch_definition=batch_definition
+            )
+            batch_spec_options = {
+                "reader_method": self._get_reader_method(),
+                "reader_options": self.dict(
+                    include=self._get_reader_options_include(),
+                    exclude=self._EXCLUDE_FROM_READER_OPTIONS,
+                    exclude_unset=True,
+                    by_alias=True,
+                ),
+            }
+            batch_spec.update(batch_spec_options)
+
+            batch_data, batch_markers = execution_engine.get_batch_data_and_markers(
+                batch_spec=batch_spec
+            )
+
+            batch_metadata = copy.deepcopy(batch_request.options)
+            # TODO: <Alex>ALEX-FIX_TO_INSURE_BASE_DIRECTORY_TYPE_IS_PATHLIB.PATH-CONSISTENTLY</Alex>
+            # TODO: <Alex>ALEX</Alex>
+            # TODO: <Alex>ALEX-FIX_TO_INSURE_PROPERTY_NAME_CORRESPONDDS_TO_ITS_MEANING</Alex>
+            batch_metadata["base_directory"] = pathlib.Path(batch_spec["path"])
+            # TODO: <Alex>ALEX</Alex>
+            # TODO: <Alex>ALEX</Alex>
+            batch_metadata.update(batch_definition.batch_identifiers)
+            batch_request.options.update(batch_definition.batch_identifiers)
+            # TODO: <Alex>ALEX</Alex>
+
+            # Some pydantic annotations are postponed due to circular imports.
+            # Batch.update_forward_refs() will set the annotations before we
+            # instantiate the Batch class since we can import them in this scope.
+            Batch.update_forward_refs()
+
+            batch = Batch(
+                datasource=self.datasource,
+                data_asset=self,
+                batch_request=batch_request,
+                data=batch_data,
+                metadata=batch_metadata,
+                legacy_batch_markers=batch_markers,
+                legacy_batch_spec=batch_spec,
+                legacy_batch_definition=batch_definition,
+            )
+            batch_list.append(batch)
+
+        # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+        # TODO: <Alex>ALEX-MOVE_SORTING_INTO_FILE_PATH_DATA_CONNECTOR_ON_BATCH_DEFINITION_OBJECTS</Alex>
+        self.sort_batches(batch_list)
+
+        return batch_list
+
     def test_connection(self) -> None:
         """Test the connection for the DataAsset.
 
         Raises:
             TestConnectionError: If the connection test fails.
         """
-        raise NotImplementedError(
-            """One needs to implement "test_connection" on a _FilePathDataAsset subclass."""
-        )
+        data_connector: DataConnector = self._get_data_connector()
```

Comment:
@donaldheppner Please note how elegantly `test_connection()` is now implemented for all file-type `DataAsset` subclasses.  In fact, once the SQL DataAsset implementations are also updated to utilize a `DataConnector`, then this code and be moved one-level up (to the `DataAsset` parent class), thereby allowing us to delete all the lower-level code.  Thank you.